gamemode/modules/oceanitems/sv_oceanitems.lua

```diff
@@ -0,0 +1,141 @@
+        end
+    end
+
+    if item.itemType then
+        item.class = "rvr_dropped_item"
+        item.beforeSpawn = function( ent )
+            local count = math.random( item.mincount or 1, item.maxcount  or 1 )
+            ent:Setup( RVR.Items.getItemData( item.itemType ), count )
+        end
+    end
+end
+
+
+local function calculateCumulativeWeights( tbl )
+    for i = 1, #tbl do
+        local item = tbl[i]
+        local previous = tbl[i-1]
+        local previousWeight = previous and previous.weight or 0
+
+        item.cumWeight = item.weight + previousWeight
+    end
+end
+
+local function getRandomWeightedValue( tbl )
+    local lastItem = tbl[#tbl]
+    if not lastItem.cumWeight then
+        calculateCumulativeWeights( tbl )
+    end
+
+    local randNum = math.random( 1, lastItem.cumWeight )
+
+    for _, item in next, tbl do
+        if randNum <= item.cumWeight then return item end
+    end
+    return lastItem
+end
+
+
+function getRandomBoxItems( amount )
+    local items = {}
+    for i=1, amount do
+        local item = getRandomWeightedValue( config.SCRAP_BARREL_ITEMS )
+        local count = math.random( item.mincount or 1, item.maxcount or 1 )
+        table.insert( items, { itemType = item.itemType, count = count } )
+    end
+
+    return items
+end
+
+local function getRandomPosition( origin, innerRadius, width )
+    local randomAngle = math.random() * 2 * math.pi
+    local radius = innerRadius + math.random() * width
+
+    local x, y = math.cos( randomAngle ) * radius, math.sin( randomAngle ) * radius
+    return origin + Vector( x, y, 0 )
+end
+
+local function getRandomPly()
+    local plys = player.GetHumans()
+    local alivePlayers = {}
+    for _, ply in pairs( plys ) do
+        if ply:Alive() then
+            alivePlayers[#plys+1] = ply
+        end
+    end
+    if #alivePlayers == 0 then return nil end
+
+    return plys[math.random( 1, #plys )]
+end
+
+
+RVR.Trash.spawnedTrashList = {}
+
+local function createTrashForPlayer( ply )
+    local pos = getRandomPosition( ply:GetPos(), config.SPAWN_RADIUS, config.SPAWN_WIDTH )
+    pos.z = RVR.waterSurfaceZ + config.SPAWN_Z_OFFSET
+
+    if not util.IsInWorld( pos ) then return end
+
+    local item = getRandomWeightedValue( config.POSSIBLE_ITEMS )
+    if not item.class then return  end
+
+    local ent = ents.Create( item.class )
+    ent:SetPos( pos )
+
+    if item.beforeSpawn then item.beforeSpawn( ent ) end
+    ent:Spawn()
+    if item.afterSpawn then item.afterSpawn( ent ) end
+
+    table.insert( RVR.Trash.spawnedTrashList, ent )
+end
+
+local function createTrash( amount )
+    for i=1, amount do
+        local ply = getRandomPly()
+
+        if not ply then break end
+
+        createTrashForPlayer( ply )
+    end
+end
+
+timer.Create( "RVR_CreateTrash", 5, 0, function()
+    local amountPerPlayer = config.MAX_TRASH_PER_PlAYER
+    local amount = amountPerPlayer * #player.GetHumans() - #RVR.Trash.spawnedTrashList
```

Comment:
What if one of these trash items is removed then it wont be sequential anymore?

Reply:
yes it will, when an entity is removed it becomes NULL not nil.